packages/autorest.typescript/src/utils/autorestOptions.ts

```diff
@@ -73,10 +74,16 @@ export async function extractAutorestOptions(): Promise<AutorestOptions> {
     azureSdkForJs,
     productDocLink,
     coreHttpCompatMode,
-    dependencyInfo
+    dependencyInfo,
+    useLegacyLro
   };
 }
 
+async function getUseLegacyLro(host: AutorestExtensionHost): Promise<boolean> {
+  const useLegacyLroOption = await host.getValue("use-legacy-lro");
+  return useLegacyLroOption === null ? false : Boolean(useLegacyLroOption);
```

Comment:
Enabling this flag will generate the old LRO implementation. However, the new LRO implementation, added here, is not backward compatible so this flag shouldn't be used to re-generate already GA'd libraries.

The flag is disabled by default, so to re-generate already GA'd libraries, the flag will have to be set explicitely in the generation command.

This experience is not the best for customers of the code generator, but I believe it will help us migrate to the new implementation faster. If that doesn't work, I am open to set it to true by default.

/cc @joheredi, @qiaozha, @MaryGao

Reply:
I talked to @xirzec offline about it and he is ok with the idea in principle. I wonder if anyone else has any other thoughts?